agent/ecs_meta_data.go

```diff
@@ -0,0 +1,32 @@
+package agent
+
+import (
+	"context"
+	metadata "github.com/brunoscheufler/aws-ecs-metadata-go"
+	"net/http"
+)
+
+type ECSMetadata struct {
+}
+
+func (e ECSMetadata) Get() (map[string]string, error) {
+	metaData := make(map[string]string)
+
+	ecsMeta, err := metadata.GetContainer(context.Background(), &http.Client{})
+	if err != nil {
+		return metaData, err
+	}
+
+	switch m := ecsMeta.(type) {
+	case *metadata.ContainerMetadataV3:
+		metaData["ecs:container-name"] = m.DockerName
+		metaData["ecs:image"] = m.Image
+		metaData["ecs:task-arn"] = m.Labels.EcsTaskArn
+	case *metadata.ContainerMetadataV4:
+		metaData["ecs:container-name"] = m.DockerName
+		metaData["ecs:image"] = m.Image
+		metaData["ecs:task-arn"] = m.Labels.EcsTaskArn
+	}
```

Comment:
looking at the code in [aws-ecs-metadata-go](https://github.com/brunoscheufler/aws-ecs-metadata-go"), it shouldn't be possible for it to return anything other than `ContainerMetadataV3` or `ContainerMetadataV4`, but it might still be worth adding 

```suggestion
	default:
		return fmt.Errorf("ecs metadata returned unknown type %T", m)
	}
```

just in case something changes